src/e2e-test/java/io/cdap/plugin/common/stepsdesign/TestSetupHooks.java

```diff
@@ -1030,4 +1033,17 @@ public static void createSinkBQExistingDatatypeTable() throws IOException, Inter
     PluginPropertyUtils.addPluginProp(" bqTargetTable",  bqTargetTable);
     BeforeActions.scenario.write("BQ Target Table " +  bqTargetTable + " updated successfully");
   }
+  private static String createGCSBucketWithFileAndLifeCycle(String filePath) throws IOException, URISyntaxException {
+    String bucketName = StorageClient.createBucketwithLifeCycle("00000000-e2e-" + UUID.randomUUID(),5 ).getName();
+    StorageClient.uploadObject(bucketName, filePath, filePath);
+    PluginPropertyUtils.addPluginProp("gcsSourceBucketName", bucketName);
+    PluginPropertyUtils.addPluginProp("gcsSourcePath", "gs://" + bucketName + "/" + filePath);
+    BeforeActions.scenario.write("Created GCS Bucket " + bucketName + " containing " + filePath + " file");
+    return bucketName;
+  }
+
+  @Before(order = 1, value = "@GCS_CSV_MULTI_PART_UPLOAD")
+  public static void createBucketWithCSVFileAND() throws IOException, URISyntaxException {
+    gcsTargetBucketName = createGCSBucketWithFileAndLifeCycle(PluginPropertyUtils.pluginProp("gcsCsvFile"));
```

Comment:
Why are we uploading a civ file here in bucket? the pipeline from BQ to GCS itself should create a file in bucket.

Reply:
createGCSBucketWithFileAndLifeCycle this method required the file path , so this is the CSV file path.

Reply:
in its parameter it is required.

Reply:
Changes Added as per the Suggestion.